LibC/stdio.cpp

```diff
@@ -293,6 +293,8 @@ void rewind(FILE* stream)
 
 int dbgprintf(const char* fmt, ...)
 {
+    // if this fails, you're printing too early.
+    ASSERT(stddbg);
```

Comment:
Great assertion. This is a bit unfortunate, maybe we should go through some extra hoops to make dbgprintf() always work.. unrelated to this patch ofc.